cmd/kube-proxy/app/server.go

```diff
@@ -780,17 +783,31 @@ func (s *ProxyServer) birthCry() {
 	s.Recorder.Eventf(s.NodeRef, api.EventTypeNormal, "Starting", "Starting kube-proxy.")
 }
 
-func getNodeIP(client clientset.Interface, hostname string) net.IP {
+func getNodeIP(client clientset.Interface, hostname string) (net.IP, error) {
 	var nodeIP net.IP
-	node, err := client.Core().Nodes().Get(hostname, metav1.GetOptions{})
-	if err != nil {
-		glog.Warningf("Failed to retrieve node info: %v", err)
-		return nil
+	var complained bool
+	tryGet := func() (bool, error) {
+		node, err := client.Core().Nodes().Get(hostname, metav1.GetOptions{})
+		if err != nil {
+			glog.Warningf("Failed one attempt to retrieve node info: %v", err)
+			complained = true
+			return false, nil
+		}
+		nodeIP, err = utilnode.InternalGetNodeHostIP(node)
+		if err != nil {
+			return false, fmt.Errorf("Failed to retrieve IP of node: %v", err)
+		}
+		return true, nil
 	}
-	nodeIP, err = utilnode.InternalGetNodeHostIP(node)
-	if err != nil {
-		glog.Warningf("Failed to retrieve node IP: %v", err)
-		return nil
+	backoff := wait.Backoff{
+		Duration: time.Second * 10,
+		Factor:   2.0,
+		Jitter:   0.25,
+		Steps:    6,
+	}
+	err := wait.ExponentialBackoff(backoff, tryGet)
```

Comment:
Not a huge problem at all, but it's usually more idomatic to put the function inside the ExponentialBackoff, like:

```
var complained bool
err := wait.ExponentialBackoff(backoff, func() (bool, error) {
    node, err := client.Core().Nodes().Get(hostname, metav1.GetOptions{})
    <all the rest>
})
if complained && err == nil {
```